.tree-sitter-lint/tree-sitter-lint-local/src/lib.rs

```diff
@@ -26,15 +26,36 @@ pub fn run_fixing_for_slice<'a>(
 ) -> Vec<ViolationWithContext> {
     tree_sitter_lint::run_fixing_for_slice(file_contents, tree, path, args_to_config(args))
 }
-
+struct LocalLinterConcrete;
+impl LocalLinter for LocalLinterConcrete {
```

Comment:
To "inject" the "custom local" behavior needed for the LSP server this is what I came up with (a trait that effectively is a proxy to the custom local `tree_sitter_lint` entry points)

(vs generating a custom "giant blob" of all of the LSP code with these particular custom local "entry points" hard-linked in - in general I'm following the rule of thumb of trying to keep the generated code as minimal/DRY as possible and pushing the "real logic" into the actual main repo)